src/Microsoft.AspNetCore.Mvc.ViewFeatures/Rendering/HtmlHelperInputExtensions.cs

```diff
@@ -762,47 +531,26 @@ public static IHtmlContent TextBox(this IHtmlHelper htmlHelper, string expressio
         }
 
         /// <summary>
-        /// Returns an &lt;input&gt; element of type "text" for the specified <paramref name="expression"/>.
+        /// Returns an &lt;input&gt; element of type "text" for the specified <paramref name="expression"/>. Adds a
+        /// "value" attribute to the element containing the first non-<c>null</c> value found in:
+        /// the <see cref="ActionContext.ModelState"/> entry with full name,
+        /// the <paramref name="value"/> parameter,
+        /// the <see cref="IHtmlHelper.ViewData"/> entry with full name, or
+        /// the <paramref name="expression"/> evaluated against <see cref="ViewFeatures.ViewDataDictionary.Model"/>.
+        /// See <see cref="IHtmlHelper.Name"/> for more information about a "full name".
         /// </summary>
         /// <param name="htmlHelper">The <see cref="IHtmlHelper"/> instance this method extends.</param>
         /// <param name="expression">Expression name, relative to the current model.</param>
         /// <param name="value">If non-<c>null</c>, value to include in the element.</param>
         /// <param name="format">
-        /// The composite format <see cref="string"/> (see http://msdn.microsoft.com/en-us/library/txafckwd.aspx).
+        /// The composite format <see cref="string"/> (see http://msdn.microsoft.com/en-us/library/txafckwd.aspx) used
```

Comment:
Change to: `The format string (see HTTPS://<url>) to format ...`

In particular:

1. No one has ever heard of the term "composite format" (I had to look it up), so let's use the common terminology of "format string"
2. The term "string" here isn't the type name, so it shouldn't be a link
3. Always use HTTPS URLs when available (and for MSDN it's always available).

Please apply this elsewhere in the PR as needed.